cmd/utils.go

```diff
@@ -272,15 +273,54 @@ func getProjectConfig(config *RootCommandConfig) (ProjectConfig, error) {
 
 		}
 		appsodyConfig := filepath.Join(dir, ConfigFile)
+
 		v := viper.New()
 		v.SetConfigFile(appsodyConfig)
 		Debug.log("Project config file set to: ", appsodyConfig)
+
 		err := v.ReadInConfig()
+
 		if err != nil {
 			var tempProjectConfig ProjectConfig
 			return tempProjectConfig, errors.Errorf("Error reading project config %v", err)
 
 		}
+
+		if prjName != "" {
+			prjName = strings.ToLower(prjName)
+			r, _ := regexp.Compile("[^a-z0-9]+")
+			prjName = r.ReplaceAllString(prjName, "-")
+			match, _ := regexp.MatchString("[a-z]([-a-z0-9]*[a-z0-9])?", prjName)
+
+			if !match {
+				return *config.ProjectConfig, errors.Errorf("This is not a valid project name")
+			}
+
+			v.Set("project-name", prjName)
+			err := v.WriteConfig()
+			if err != nil {
+				Error.log(err)
+			}
+
+			Info.log("Your project name is ", prjName)
+		} else {
+			project := "appsody-" + strings.ToLower(filepath.Base(dir)) + "-app"
+			reg, err := regexp.Compile("[^a-z0-9]+")
+			if err != nil {
+				Error.log(err)
+			}
+			projectName := reg.ReplaceAllString(project, "-")
```

Comment:
Even after the substitution, we should check that the name is still compliant with the K8S regex. I understand that it should be at this point - but better double check.

Reply:
changed